Makefile

```diff
@@ -10,21 +10,31 @@ MOCKGEN := $(GOPATH)/bin/mockgen
 $(MOCKGEN):
 	go get -u github.com/golang/mock/mockgen
 
+cleanup:
+	rm -rf $(GOPATH)/src/$(PKG)/samples/gen-js/*
+	rm -rf $(GOPATH)/src/$(PKG)/samples/gen-js-no-definitions/*
+	rm -rf $(GOPATH)/src/$(PKG)/samples/gen-js-deprecated/*
+	rm -rf $(GOPATH)/src/$(PKG)/samples/gen-js-wag-patch/*
+	rm -rf $(PKG)/samples/gen-go/*
+	rm -rf $(PKG)/samples/gen-go-no-definitions/*
+	rm -rf $(PKG)/samples/gen-go-deprecated/*
+	rm -rf $(PKG)/samples/gen-go-wag-patch/*
```

Comment:
There's probably a better way to do this, but added this to ensure that the samples folder represents the _current_ state of the world. For an example (that I encountered) if we update the client gen to stop creating a file it may still stay committed into the samples directory.